## Imports & Setup

In [1]:
import pandas as pd
import numpy as np

pd.set_option("display.max_columns", None)
pd.set_option("display.max_rows", 50)


## Raw files from the Data Mart

In [3]:
file1 = "2014_2016_student_success_services.csv"
file2 = "2017_2020_student_success_services.csv"
file3 = "2021_2025_student_success_services.csv"

df1 = pd.read_csv(file1)
df2 = pd.read_csv(file2)
df3 = pd.read_csv(file3)


## Inspect the First Few Rows

In [4]:
df1.head(10)


Unnamed: 0                                         Unnamed: 1  \
0                NaN                                                NaN   
1                NaN                                                NaN   
2  San Francisco CCD                                                NaN   
3                NaN  Credit - Academic/Progress Probation Services ...   
4                NaN  Credit - Counseling/Advisement Services       ...   
5                NaN  Credit - Education Plan Services              ...   
6                NaN  Credit - Initial Assessment Services Placement...   
7                NaN  Credit - Initial Orientation Services         ...   
8                NaN  Credit - Other Services                       ...   
9                NaN  Non-Credit - Counseling/Advisement Services   ...   

                                     Summer 2014  \
0                 Directed                         
1  Service Received                                
2                                            NaN   
3                                              9   
4                                          1,659   
5                                          1,534   
6                                            784   
7                                            431   
8                                          2,091   
9                                              0   

                                   Summer 2014.1  \
0                 Directed                         
1  Service Not Received                            
2                                            NaN   
3                                            NaN   
4                                          8,312   
5                                          8,437   
6                                          8,233   
7                                          8,639   
8                                            NaN   
9                                              0   

                                   Summer 2014.2  \
0                 Exempted                         
1  Service Received                                
2                                            NaN   
3                                            NaN   
4                                              0   
5                                              0   
6                                             49   
7                                             17   
8                                            NaN   
9                                              0   

                                   Summer 2014.3  \
0                 Exempted                         
1  Service Not Received                            
2                                            NaN   
3                                            NaN   
4                                              0   
5                                              0   
6                                            905   
7                                            884   
8                                            NaN   
9                                              0   

                                       Fall 2014  \
0                 Directed                         
1  Service Received                                
2                                            NaN   
3                                             16   
4                                          5,973   
5                                          3,860   
6                                          3,858   
7                                          2,953   
8                                          6,123   
9                                              0   

                                     Fall 2014.1  \
0                 Directed                         
1  Service Not Received                            
2                                            NaN   
3                                            NaN   
4                                         53,990   
5      

In [5]:
df2.head(10)
df3.head(10)


Unnamed: 0                                         Unnamed: 1  \
0                NaN                                                NaN   
1                NaN                                                NaN   
2  San Francisco CCD                                                NaN   
3                NaN  Credit - Academic/Progress Probation Services ...   
4                NaN  Credit - Counseling/Advisement Services       ...   
5                NaN  Credit - Education Plan Services              ...   
6                NaN  Credit - Initial Assessment Services Placement...   
7                NaN  Credit - Initial Orientation Services         ...   
8                NaN  Credit - Other Services                       ...   
9                NaN  Non-Credit - Counseling/Advisement Services   ...   

                                     Spring 2021  \
0                 Directed                         
1  Service Received                                
2                                            NaN   
3                                             61   
4                                          6,067   
5                                          4,379   
6                                          3,663   
7                                          1,216   
8                                          5,663   
9                                              0   

                                   Spring 2021.1  \
0                 Directed                         
1  Service Not Received                            
2                                            NaN   
3                                            NaN   
4                                         15,997   
5                                         17,685   
6                                         17,405   
7                                         19,744   
8                                            NaN   
9                                              5   

                                   Spring 2021.2  \
0                 Exempted                         
1  Service Received                                
2                                            NaN   
3                                            NaN   
4                                          1,800   
5                                          1,158   
6                                          1,509   
7                                            643   
8                                            NaN   
9                                            126   

                                   Spring 2021.3  \
0                 Exempted                         
1  Service Not Received                            
2                                            NaN   
3                                            NaN   
4                                          7,129   
5                                          7,771   
6                                          8,416   
7                                          9,390   
8                                            NaN   
9                                         30,862   

                                     Summer 2021  \
0                 Directed                         
1  Service Received                                
2                                            NaN   
3                                             10   
4                                          1,695   
5                                          1,467   
6                                          1,419   
7                                              5   
8                                          2,032   
9                                              0   

                                   Summer 2021.1  \
0                 Directed                         
1  Service Not Received                            
2                                            NaN   
3                                            NaN   
4                                          9,020   
5      

## Drop Metadata Rows

In [6]:
df1_clean = df1.iloc[3:].reset_index(drop=True)
df2_clean = df2.iloc[3:].reset_index(drop=True)
df3_clean = df3.iloc[3:].reset_index(drop=True)


In [7]:
df1_clean.head() #for inspection


Unnamed: 0                                         Unnamed: 1 Summer 2014  \
0        NaN  Credit - Academic/Progress Probation Services ...           9   
1        NaN  Credit - Counseling/Advisement Services       ...       1,659   
2        NaN  Credit - Education Plan Services              ...       1,534   
3        NaN  Credit - Initial Assessment Services Placement...         784   
4        NaN  Credit - Initial Orientation Services         ...         431   

  Summer 2014.1 Summer 2014.2 Summer 2014.3 Fall 2014 Fall 2014.1 Fall 2014.2  \
0           NaN           NaN           NaN        16         NaN         NaN   
1         8,312             0             0     5,973      53,990           0   
2         8,437             0             0     3,860      56,103           0   
3         8,233            49           905     3,858      52,675         116   
4         8,639            17           884     2,953      53,773          52   

  Fall 2014.3 Spring 2015 Spring 2015.1 Spring 2015.2 Spring 2015.3  \
0         NaN         152           NaN           NaN           NaN   
1           0       7,908        49,810             0             0   
2           0       6,025        51,693             0             0   
3       3,314       3,088        51,254           114         3,262   
4       3,185       2,159        52,384            35         3,140   

  Summer 2015 Summer 2015.1 Summer 2015.2 Summer 2015.3 Fall 2015 Fall 2015.1  \
0          71           NaN           NaN           NaN       176         NaN   
1       3,453        14,638             0             0    13,047      46,747   
2       2,787        15,304             0             0     9,097      50,697   
3       1,006        16,245            31           809     4,529      52,121   
4         754        16,571            12           754     3,539      52,954   

  Fall 2015.2 Fall 2015.3 Spring 2016 Spring 2016.1 Spring 2016.2  \
0         NaN         NaN         102           NaN           NaN   
1           0           0      11,637        45,932             0   
2           0           0       8,405        49,164             0   
3         106       3,038       4,375        50,090           115   
4          64       3,237       2,824        51,517            47   

  Spring 2016.3 Summer 2016 Summer 2016.1 Summer 2016.2 Summer 2016.3  \
0           NaN          29           NaN           NaN           NaN   
1             0       5,462        19,613             0             0   
2             0       4,623        20,452             0             0   
3         2,989       3,242        20,804            71           958   
4         3,181       1,910        22,093            23         1,049   

  Fall 2016 Fall 2016.1 Fall 2016.2 Fall 2016.3  
0        12         NaN         NaN         NaN  
1     5,348      38,815           0           0  
2     2,013      42,150           0           0  
3     3,741      38,024          79       2,319  
4     3,304      38,186          54       2,619

## Rename the Service Column

In [8]:
df1_clean.rename(columns={df1_clean.columns[1]: "service"}, inplace=True)
df2_clean.rename(columns={df2_clean.columns[1]: "service"}, inplace=True)
df3_clean.rename(columns={df3_clean.columns[1]: "service"}, inplace=True)


In [9]:
df1_clean["service"].head()


0    Credit - Academic/Progress Probation Services ...
1    Credit - Counseling/Advisement Services       ...
2    Credit - Education Plan Services              ...
3    Credit - Initial Assessment Services Placement...
4    Credit - Initial Orientation Services         ...
Name: service, dtype: object

## Clean and Simplified Service Names

In [12]:
def clean_service_name(s):
    if pd.isna(s):
        return s
    s = s.lower()
    if "counsel" in s:
        return "counseling"
    if "education plan" in s or "ed plan" in s:
        return "ed_plan"
    if "orientation" in s:
        return "orientation"
    if "assessment" in s:
        return "assessment"
    if "probation" in s:
        return "probation"
    return s.strip()

df1_clean["service"] = df1_clean["service"].apply(clean_service_name)
df2_clean["service"] = df2_clean["service"].apply(clean_service_name)
df3_clean["service"] = df3_clean["service"].apply(clean_service_name)



In [13]:
df1_clean["service"].unique()


array(['probation', 'counseling', 'ed_plan', 'assessment', 'orientation',
       'credit - other services', 'non-credit - other services'],
      dtype=object)

### I will only keep "Service Received" Columns

In [14]:
cols_df1 = [c for c in df1_clean.columns if ".1" in c] 
cols_df2 = [c for c in df2_clean.columns if ".1" in c]
cols_df3 = [c for c in df3_clean.columns if ".1" in c]

df1_received = df1_clean[["service"] + cols_df1]
df2_received = df2_clean[["service"] + cols_df2]
df3_received = df3_clean[["service"] + cols_df3]


In [15]:
df1_received.head()


service Summer 2014.1 Fall 2014.1 Spring 2015.1 Summer 2015.1  \
0    probation           NaN         NaN           NaN           NaN   
1   counseling         8,312      53,990        49,810        14,638   
2      ed_plan         8,437      56,103        51,693        15,304   
3   assessment         8,233      52,675        51,254        16,245   
4  orientation         8,639      53,773        52,384        16,571   

  Fall 2015.1 Spring 2016.1 Summer 2016.1 Fall 2016.1  
0         NaN           NaN           NaN         NaN  
1      46,747        45,932        19,613      38,815  
2      50,697        49,164        20,452      42,150  
3      52,121        50,090        20,804      38,024  
4      52,954        51,517        22,093      38,186

### Converting strings

In [18]:
for df in [df1_received, df2_received, df3_received]:
    for c in df.columns:
        if c != "service":
            df.loc[:, c] = (
                df[c]
                .astype(str)
                .str.replace(",", "", regex=False)
                .replace("nan", np.nan)
                .astype(float)
            )


## Reshape Wide → Long Format

In [19]:
df1_long = df1_received.melt(id_vars="service",
                             var_name="term_year",
                             value_name="count")
df2_long = df2_received.melt(id_vars="service",
                             var_name="term_year",
                             value_name="count")
df3_long = df3_received.melt(id_vars="service",
                             var_name="term_year",
                             value_name="count")


In [20]:
df1_long.head()


service      term_year   count
0    probation  Summer 2014.1     NaN
1   counseling  Summer 2014.1  8312.0
2      ed_plan  Summer 2014.1  8437.0
3   assessment  Summer 2014.1  8233.0
4  orientation  Summer 2014.1  8639.0

## Convert NaN to Zero After Melt

In [21]:
df1_long["count"] = df1_long["count"].fillna(0)
df2_long["count"] = df2_long["count"].fillna(0)
df3_long["count"] = df3_long["count"].fillna(0)


In [22]:
df1_long.head()

service      term_year   count
0    probation  Summer 2014.1     0.0
1   counseling  Summer 2014.1  8312.0
2      ed_plan  Summer 2014.1  8437.0
3   assessment  Summer 2014.1  8233.0
4  orientation  Summer 2014.1  8639.0

## Extract Term & Year

In [23]:
def split_term_year(s):
    parts = s.split()
    term = parts[0]
    year = parts[1].replace(".1", "")
    return term, int(year)

df1_long[["term", "year"]] = df1_long["term_year"].apply(
    lambda x: pd.Series(split_term_year(x))
)
df2_long[["term", "year"]] = df2_long["term_year"].apply(
    lambda x: pd.Series(split_term_year(x))
)
df3_long[["term", "year"]] = df3_long["term_year"].apply(
    lambda x: pd.Series(split_term_year(x))
)


## Master Dataset

In [24]:
master = pd.concat([df1_long, df2_long, df3_long], ignore_index=True)

master = master[["year", "term", "service", "count"]].dropna()

master.head()


year    term      service   count
0  2014  Summer    probation     0.0
1  2014  Summer   counseling  8312.0
2  2014  Summer      ed_plan  8437.0
3  2014  Summer   assessment  8233.0
4  2014  Summer  orientation  8639.0

## Final Dataset 

In [25]:
master.to_csv("cleaned_student_success_services.csv", index=False)
master.head()


year    term      service   count
0  2014  Summer    probation     0.0
1  2014  Summer   counseling  8312.0
2  2014  Summer      ed_plan  8437.0
3  2014  Summer   assessment  8233.0
4  2014  Summer  orientation  8639.0

## Data Cleaning & Standardization Pipeline Summary

To prepare the Student Success Services dataset for analysis, I built a multi-step cleaning pipeline that standardizes all DataMart files from 2014–2025 into a single structured dataset. First, I loaded each CSV and removed the top metadata rows that contained descriptive labels instead of data. Then, I extracted the true service names and standardized them (e.g., “Credit – Counseling/Advisement Services” → “counseling”). Because DataMart reports each term using four parallel columns (Directed, Service Received, Exempted, Not Received), I selected only the “Service Received” columns for analysis and converted student counts from string values (e.g., “8,312”) into numeric format. I reshaped each dataset from wide format to long format using melt(), resulting in one row per service per term. After converting blank cells to zero and parsing the term and year from column names, I merged all years into one master dataset with consistent columns: year, term, service, count. This cleaned dataset now supports accurate EDA, visualizations, and modeling across the full 2014–2025 time period.